src/circuit/circuit_frame.rs

```diff
@@ -2533,6 +2761,11 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
         let sum = constraints::add(&mut cs.namespace(|| "sum"), a, b)?;
         let diff = constraints::sub(&mut cs.namespace(|| "difference"), a, b)?;
         let product = constraints::mul(&mut cs.namespace(|| "product"), a, b)?;
+        let secret = match arg1.hash().get_value() {
+            Some(s) => s,
+            None => F::zero(), //dummy
```

Comment:
Why is this marked dummy? The default secret is zero when committing without an explicit secret — but I am not sure I would categorize that as a 'dummy' value.

Reply:
Here we are dealing with binary operations, then `arg1.hash().get_value()` is `None` means we can't find the `secret` parameter, but we were supposed to find it if it was the case. Then we have that `None` happens as a dummy, because in fact it is not a `binop`. Indeed we default the secret to zero, but it happens in the unary operations part of the circuit.

Reply:
I see (took a while for me to trace through all this). In that case, I understand and agree with the comment. This might be a perfect place for the proposed dummy-tagged value (i.e. a specially-tagged value for use in circuit positions which can never be part of a real computation).